src/NuGet.Core/NuGet.Protocol/Utility/CachingUtility.cs

```diff
@@ -23,13 +25,15 @@ public static string ComputeHash(string value, bool addIdentifiableCharacters =
         {
             var trailing = value.Length > 32 ? value.Substring(value.Length - 32) : value;
             byte[] hash;
-            using (var sha = SHA1.Create())
+            using (var sha = SHA256.Create())
             {
                 hash = sha.ComputeHash(Encoding.UTF8.GetBytes(value));
             }
 
             const string hex = "0123456789abcdef";
-            return hash.Aggregate(addIdentifiableCharacters ? "$" + trailing : string.Empty, (result, ch) => "" + hex[ch / 0x10] + hex[ch % 0x10] + result);
+            return hash
+                .Take(HashLength)
+                .Aggregate(addIdentifiableCharacters ? "$" + trailing : string.Empty, (result, ch) => "" + hex[ch / 0x10] + hex[ch % 0x10] + result);
```

Comment:
Wow, mega string allocations!  I wouldn't be surprised if optimising this would make a measurable (but probably still small, but still, actually measurable) performance gain.

Reply:
`ConcurrencyUtilities` already has a `ToHex` method that does 2 allocations, not 40. Please copy that method here.

Reply:
Thanks! Fixed.